pinot-segment-local/src/main/java/org/apache/pinot/segment/local/utils/HashUtils.java

```diff
@@ -44,8 +73,30 @@ public static Object hashPrimaryKey(PrimaryKey primaryKey, HashFunction hashFunc
         return new ByteArray(HashUtils.hashMD5(primaryKey.asBytes()));
       case MURMUR3:
         return new ByteArray(HashUtils.hashMurmur3(primaryKey.asBytes()));
+      case UUID:
+        return new ByteArray(HashUtils.hashUUID(primaryKey.getValues()));
       default:
         throw new IllegalArgumentException(String.format("Unrecognized hash function %s", hashFunction));
     }
   }
+
+  /**
+   * Concatenates the string representation of all values into a single byte array. Each element is appended with a
+   * byte-0 delimiter.
+   */
+  private static byte[] concatenate(Object[] values) {
+    byte[][] allValueBytes = new byte[values.length][];
+    int totalLen = 0;
+    for (int j = 0; j < allValueBytes.length; j++) {
+      allValueBytes[j] = values[j] == null ? "null\0".getBytes(StandardCharsets.UTF_8)
+          : (values[j].toString() + "\0").getBytes(StandardCharsets.UTF_8);
+      totalLen += allValueBytes[j].length;
```

Comment:
minor one: Do we need to append byte-0 to the last element?

Reply:
Had thought about it while implementing. I did it for simplicity since we don't really gain much by skipping last element, and it increases cyclomatic complexity slightly.